### PR TITLE
Update to latest `rust-dlc`

### PR DESCRIPTION
**Pull request by luckysori**
Other than breaking changes to certain `rust-dlc` APIs, the main hurdle was to fix errors on `ln-dlc-node` tests that built on first
opening a DLC channel.

To do so we had to manually instruct the channel offerer to process the `Finalize` message. Before the dependency update this wasn't necessary, but now one must finish the channel open protocol before doing other things with the channel (such as closing it collaboratively), which makes sense.

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -107,8 +107,14 @@ wipe-app:
...
 
 clippy:
-    cd mobile/native && cargo clippy --all-targets -- -D warnings
-    cd coordinator && cargo clippy --all-targets -- -D warnings
+    cd mobile/native && just cargo-clippy
+    cd coordinator && just cargo-clippy
+    cd maker && just cargo-clippy
+    for crate in crates/*; do (cd "$crate" && just cargo-clippy); done
+
+[private]
```
Comment (klochowicz) [inline justfile:115]: nice one, didn't know about `[private]`!

File: justfile [hunk 2]
```diff
@@ -144,6 +144,9 @@ native-test:
 
 test: flutter-test native-test
 
+ln-dlc-node-test:
```
Comment (klochowicz) [inline justfile:147]: perhaps a comment explaining why this one has its own recipe? e.g. `# slow tests, requiring `just docker` to be active`

**Review by klochowicz — APPROVED** [review]
LGTM!

**Comment by luckysori** [comment]
bors retry

**Comment by luckysori** [comment]
bors retry

**Review by holzeis — APPROVED** [review]

**Comment by luckysori** [comment]
bors retry

**Pull request merged**